### DISCUSSION TITLE
Using "Run" in Server UI in conjunction with a scheduled flow

### DISCUSSION THREAD

**Discussion by ReubenTheDS** [PrefectHQ/prefect#3867] [General]
Hi Team, I would like your inputs on using Runs triggered in Prefect Server UI when the Flow is a scheduled one.

**Background:**
1) I have a Prefect Server running on localhost:8080, along with a local agent & a project "ProjX" setup using the appropriate commands
2) The flow in my  python script is scheduled to run everyday & "connects" to "ProjX" :
```
def get_todays_date():
    return f'{pendulum.now().to_date_string()}'

@task(log_stdout=True)
def task1(date):
    print(f'{date}')

@task(log_stdout=True)
def task2(res1):
    # ... other cool stuff with res1

schedule = IntervalSchedule( start_date=datetime.utcnow(), interval=timedelta(days=1))

with Flow("My_flow", schedule=schedule) as flow:
    date_to_print = Parameter("date_to_print", default=[get_todays_date()])
    res1 = task1(date_to_print)
    res2 = task2(res1)
    # more tasks here
    # and here

flow.register(project_name="ProjX")
```
I confirm I can see the scheduled flow-runs in the UI 

3) In the UI, I can trigger a flow-run to print "2020-12-25" using:
```
{'date_to_print':'2020-12-25'}
```

**The expectation:**
Every day the **scheduled flow runs**, it should print the date for that day

**The problem:**
The day printed **by scheduled flow-runs** is the first day that I set up the Prefect Server & connected to it from the script.

What am I missing, or, how could I do this better?

**Comment by cicdw** [comment]
Hi @ReubenTheDS - this is [a common confusion for users new to deferred computation](https://stackoverflow.com/a/64158793/1617887).

On this line:
```python
    date_to_print = Parameter("date_to_print", default=[get_todays_date()])
```
your `get_todays_date` function runs, and its return value is hardcoded as the `Parameter`'s default.   In your case, it might be easier to rely on [datetime values stored in Prefect context](https://docs.prefect.io/api/latest/utilities/context.html) than writing up custom logic.

**Reply by ReubenTheDS** [reply]
Thank you for the clarification . ```prefect.context``` helps
